### PR TITLE
[tune] Tune CLI Fixes

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?

- [x] Add `--limit` flag for `ls`
- [x] Add ordering functionality to `--sort` flag
- [x] Remove `last_result` from the names of columns for `ls`
- [x] Fix weird double quote error messages (`\"`)

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by andrewztan** [comment]
@richardliaw wip

File: python/ray/tune/commands.py [hunk 1]
```diff
@@ -130,35 +124,45 @@ def list_trials(experiment_path,
...
+
+    checkpoint_dicts = []
+    for g in checkpoints:
+        config = g.pop("config")
+        config = flatten_dict(config)
+        last_result = g.pop("last_result")
+        last_result = flatten_dict(last_result)
+        g = flatten_dict(g)
+        combined_dict = {**config, **last_result, **g}
+        checkpoint_dicts.append(combined_dict)
```
Comment (richardliaw) [inline python/ray/tune/commands.py]: what about something like

```python
UNNEST_KEYS = ["config", "last_result"] # global

...
def list_trials(...):
    ...
    checkpoint_dicts = []
    for g in checkpoints:
        for key in UNNEST_KEYS:
            if key not in g:
                continue
            unnest_dict = flatten_dict(g.pop(key))
            g.update(key)
        checkpoint_dicts.append(g)
```

Comment (andrewztan) [reply python/ray/tune/commands.py]: are `config` and `last_result` the only nested dicts in `g`? or should we do `g = flatten_dict(g)` right before `checkpoint_dicts.append(g)`

**Comment by richardliaw** [comment]
I think those are the ones we care about for now!

**Comment by andrewztan** [comment]
@richardliaw seems to be failing on `test_sgd.py`:
```
ray.exceptions.RayTaskError: ray_worker (pid=135, host=ba99eb6cfe81)
NameError: global name 'FileNotFoundError' is not defined
```

Seems unrelated?

**Comment by richardliaw** [comment]
Yeah unrelated

File: python/ray/tune/commands.py [hunk 2]
```diff
@@ -130,35 +126,45 @@ def list_trials(experiment_path,
...
     """
     _check_tabulate()
     experiment_state = _get_experiment_state(
         experiment_path, exit_on_fail=True)
 
-    checkpoint_dicts = experiment_state["checkpoints"]
-    checkpoint_dicts = [flatten_dict(g) for g in checkpoint_dicts]
+    checkpoints = experiment_state["checkpoints"]
+
+    # import ipdb; ipdb.set_trace()
```
Comment (richardliaw) [inline python/ray/tune/commands.py]:
```suggestion
```

File: python/ray/tune/commands.py [hunk 3]
```diff
@@ -130,35 +126,44 @@ def list_trials(experiment_path,
...
 
     if not info_keys:
         info_keys = DEFAULT_EXPERIMENT_INFO_KEYS
-    if not result_keys:
-        result_keys = DEFAULT_RESULT_KEYS
-    result_keys = ["last_result:{}".format(k) for k in result_keys]
-    col_keys = [
-        k for k in list(info_keys) + result_keys if k in checkpoints_df
-    ]
+    # result_keys = ["last_result:{}".format(k) for k in result_keys]
```
Comment (richardliaw) [inline python/ray/tune/commands.py]:
```suggestion
```

File: python/ray/tune/commands.py [hunk 4]
```diff
@@ -284,17 +296,20 @@ def list_experiments(project_path,
...
         if sort not in info_df:
-            raise KeyError("Sort Index \"{}\" not in: {}".format(
-                sort, list(info_df)))
-        info_df = info_df.sort_values(by=sort)
+            raise KeyError("{} not in: {}".format(sort, list(info_df)))
+        ascending = not desc
+        info_df = info_df.sort_values(by=sort, ascending=ascending)
+
+    if limit:
+        info_df = info_df[:limit]
```
Comment (richardliaw) [inline python/ray/tune/commands.py]: oh does this limit rows or columns?

Comment (andrewztan) [reply python/ray/tune/commands.py]: this limits rows

**Review by richardliaw — APPROVED** [review]
Looks great!

**Comment by richardliaw** [comment]
I'll merge this once tests pass.

**Comment by richardliaw** [comment]
@andrewztan this is failing lint

**Comment by richardliaw** [comment]
Nice! Thanks.

**Pull request merged**